docs/specifications/tests/Delegation-TP/delegation03.md

```diff
@@ -1,61 +1,137 @@
-   section.
-5. If there is at least one in-zone NS that has an IPv6 address, take the
-   one with the shortest name and add an AAAA record for it to the Additional
-   section.
-6. If the DNS packet after encoding is more than 512 octets then the test
-   fails.
+2. Obtain the set of name server names from the delegation of 
+   the *Child Zone* from the parent using [Method2].
 
-### Outcome(s)
+3. For each name server names obtained in the previous step:
+   1. Create an NS record with the *Child Zone* apex as owner name
+      and the name server name as RDATA.
+   2. Add the NS record to the Authority section of the DNS packet
+      created above using the message compression schema defined
+      in section 4.1.4 of [RFC 1035] where applicable.
 
-If the created packet fits into 512 octet, then the test succeeds.
+4. Using [Method1], get the parent zone of *Child Zone*.
 
-### Special procedural requirements
+5. Obtain the name server IP addresses per name server names for
+   the delegation using [Method4].
+
+6. Make a set of the name server names that resolve into at least one
+   IPv4 address ("IPv4 Name Server Set").
+
+7. If the *IPv4 Name Server Set* is not empty and all name server 
+   names in it are [In-Bailiwick of Parent] then:
+   1. Create an A record using one of the name server names and any 
+      IPv4 address.
+   2. Add the A record to the Additional section of the DNS packet 
+      created above using the message compression schema defined in 
+      section 4.1.4 of [RFC 1035].
+
+8. Make a set of the name server names that resolve into at least one
+   IPv6 address ("IPv6 Name Server Set").
+
+9. If the *IPv6 Name Server Set* is not empty and all name server 
+   names in it are [In-Bailiwick of Parent] then:
+   1. Create a AAAA record using one of the name server names and any 
+      IPv6 address.
+   2. Add the AAAA record to the Additional section of the DNS packet 
+      created above using the message compression schema defined in 
+      section 4.1.4 of [RFC 1035].
+
+10. If size of the DNS packet after encoding is larger than 512 octets 
+    then emit *[REFERRAL_SIZE_TOO_LARGE]* else emit 
+    *[REFERRAL_SIZE_OK]*.
+
+
+## Outcome(s)
+
+The outcome of this Test Case is "fail" if there is at least one message
+with the severity level *ERROR* or *CRITICAL*.
+
+The outcome of this Test Case is "warning" if there is at least one message
+with the severity level *WARNING*, but no message with severity level
+*ERROR* or *CRITICAL*.
+
+In other cases the outcome of this Test Case is "pass".
+
+Message                           | Default severity level (if message is emitted)
+:---------------------------------|:-----------------------------------
+REFERRAL_SIZE_TOO_LARGE           | WARNING
+REFERRAL_SIZE_OK                  | INFO
+
+
+## Special procedural requirements
 
 None
 
-### Intercase dependencies
+## Intercase dependencies
 
 None
 
--------
+## Terminology
```

Comment:
A better idea will be to refer the terminologies to the appropriate RFC's rather than having them here, as below:

Reply:
@sandoche2k, the terms "in-bailiwick" and "out-of-bailiwick" are central in this TC and in other TCs, but they are unknown to many readers. I want to make it easy to find the definitions by referring to both the RFC and the exact section and page where to find it, and this is the only way that I can see that I can do it. Do you have another suggestion?

Reply:
If we start explaining about the terminologies, then the specification text become quite large.  It is not possible to explain all DNS terminologies and explaining them is not the objective here.

The standard way is always  to link them to their appropriate references. For example for in and out bailiwick, the RFC on DNS terminologies is quite an appropriate reference.

Reply:
My idea is *not* to explain all DNS terminology in the terminology section, only in exception cases. If you are very much against the way I have treated "in-bailiwick" and "out-of-bailiwick" I could change it.

Reply:
I just wanted to point out that we should follow a uniform  pattern.

Reply:
@sandoche2k, good point. I will make the specifications as uniform as possible.